src/base/bittorrent/session.cpp

```diff
@@ -371,6 +371,9 @@ Session::Session(QObject *parent)
     , m_useProxy(false)
     , m_recentErroredTorrentsTimer(new QTimer(this))
 {
+    if (port() < 0)
```

Comment:
This check uses `-1` to represent a random port, but line [1145](https://github.com/qbittorrent/qBittorrent/pull/10523/files#diff-5961e6811ef7cd17f473ca94c53d5559R1145) uses `0`. Can we use `0` in both contexts, and then move this "magic number" to a local constant? The GUI doesn't accept `0` as a valid user-specified port.

Reply:
>This check uses -1 to represent a random port

To be clear, it represents an uninitialized number/port, not the same as a random port.

>The GUI doesn't accept 0 as a valid user-specified port.

I think it is better this way, usual users won't understand what binding to port 0 means and seeing tcp & udp listening on different port numbers might surprise them, so binding to port 0 only occurs when the option "Use different port on startup" is selected.